components/lib/screens/register.dart

```diff
@@ -7,100 +9,198 @@ class RegisterPage extends StatefulWidget {
-                hintText: 'Re enter chosen password',
-                labelText: 'Re enter Password',
+
+              // Password TextFormField
+              TextFormField(
+                obscureText: true,
+                decoration: const InputDecoration(
+                  hintText: 'Enter chosen password',
+                  labelText: 'Password',
+                ),
+                controller: pwdInputController,
+                inputFormatters: [new LengthLimitingTextInputFormatter(30)],
+                validator: (value) {
+                  if (value.isEmpty) {
+                    return 'This field is compulsory.';
+                  }
+                  return null;
+                },
               ),
-            inputFormatters: [new LengthLimitingTextInputFormatter(30)],
-            validator: (value) {
+
+              // Re Password TextFormField
+              TextFormField(
+                obscureText: true,
+                decoration: const InputDecoration(
+                  hintText: 'Re enter chosen password',
+                  labelText: 'Re enter Password',
+                ),
+                controller: confirmPwdInputController,
+                inputFormatters: [new LengthLimitingTextInputFormatter(30)],
+                validator: (value) {
                   if (value.isEmpty) {
                     return 'This field is compulsory.';
                   }
                   return null;
                 },
-            ), 
-            SizedBox(height: 30,),
-            Builder(
-              builder: (context) => RaisedButton(
-              elevation: 10,
-              child: Text("Register",style: TextStyle(fontSize: 20),),
-              padding: const EdgeInsets.all(14),
-              color: Colors.blue,
-              textColor: Colors.white,
-              onPressed: () {
-                if(_formKey.currentState.validate()){
-                    Scaffold.of(context).showSnackBar(SnackBar(
-                      content: Text("Form Submitted!"),
+              ),
+              SizedBox(
+                height: 30,
+              ),
+              Builder(
+                builder: (context) => RaisedButton(
+                    elevation: 10,
+                    child: Text(
+                      "Register",
+                      style: TextStyle(fontSize: 20),
                     ),
-                    );
-                }
-              }
-            ),),
-            SizedBox(height: 60,),
-          ],
-        ),
-      )
-    );
+                    padding: const EdgeInsets.all(14),
+                    color: Colors.blue,
+                    textColor: Colors.white,
+                    onPressed: () {
+                      if (_formKey.currentState.validate()) {
+                        if (pwdInputController.text ==
+                            confirmPwdInputController.text) {
+                          FirebaseAuth.instance
```

Comment:
It appears that this code is handling UI logic as well as data layer logic. Please follow single responsibility principle as this file has UI responsibility and data responsibility.

Reply:
@pravinyo are you telling me to differentiate the registration logic into a different file?

Reply:
Yes, Please separate logic for data saving in to separate class file. It is a better practice to have single class file do single task. let me know if you need help

Reply:
Okk. I am trying to do. If I need any help, then I will tell you